cli/example-ifc.config.js

```diff
@@ -13,6 +13,13 @@ module.exports = function(frameRouter) {
     {
       locale: 'en-US',
       hostRootUrl: window.location.origin,
+      registeredKeys: [
+        'a',
+        'b',
+        'actrl',
+        'ashift', 
+        'actrlshift'
```

Comment:
I believe there was some previous conversation about this, but I think a better format here is to just mimic the structure of a few non-controversial parts of the KeyboardEvent spec.  It will mean less logic to decode strings into a logical representation, and won't require folks to remember another way of representing key presses outside of the browser's API.

Something like:
```
{ 
  key: 'a', 
  shiftKey: false, 
  altKey: false, 
  ctrlKey: false, 
  metaKey: false 
}
```
or just `a` or `A` for short versions with no modifers or only shift (if we really need to support that).

Reply:
Yeah I am fine with the above structure and I even started work on that but I wanted to wait for more feedback. The reason why I am including the modifier keys very explicitly is because of interesting cross environment details. For example in Mac `alt+f` is ƒ but in Windows its just f which makes cross environmental details a nightmare if you are not overly explicit.

Reply:
That's fair, but since the host app presumably also has to handle those platform variations for native key events, wouldn't it be simpler to just have the client key events mimic the native version and let the host app use the same logic for both native events and iframe events? My goal is generally to give folks as uniform of an API as possible, even when that means mimicking an API that I don't like.  I'd rather have one interface that's just ok than two or more of varying quality.  

I would reconsider that though if things like angular/vue/etc somehow normalize browser key events behind the scenes and they are doing it consistently.

Reply:
So there are two parts to this and I think an exact mimic is going to be hard to pull off over time especially as the spec changes. The two parts are the registration and the propagation. 

In the case of the registration admittedly it is a bit easier and less error prone to limit the options to the above. The options we are leaving out are code (not yet supported in IE and Edge) and keyCode (supposedly deprecated for code). The question becomes should we be helping developers map intents by hiding some complexity? Also what do you do when the developer provides code, keyCode and key what is your actual comparison at that point? In my opinion this specific part should probably stay as the above object.

In the case of propagation back to the host I will admit that maybe we could be a bit more consistent and provide the un-translated key as well as the properties code and keyCode. I think the benefit that angular and vue have over our use is they can just provide the raw keyboard events where we cannot because we are going cross origin. The only concern I have with making this very specific to browser detail is every time the keyboard event spec changes we will have to update our implementation to send more data.

Reply:
To be clear, I'm not suggesting that we pass through or use any data besides the interface:
```
{ 
  key: string,
  shiftKey: boolean, 
  altKey: boolean, 
  ctrlKey: boolean, 
  metaKey: boolean 
}
```

keyCode, code, etc. are unnecessary for this interface at the point, and we should leave them out unless we have a specific use for them. There's no need to worry about those fields changing in the browser spec anytime in the foreseeable future.

Reading between the lines a bit, I think your primary concern is that to handle `alt+f` across OS, developers would have to register two keys to handle the case where OSX returns ƒ. Is that correct?  If so, I think it's worth noting that using the option(alt) key for OSX keyboard shortcuts breaks the OS conventions and that command is probably more appropriate in that case. As I understand it, the point of the option key is to provide a way to input alternate characters, so masking that seems strange.  Worst case, it's also not the end of the world to have to do some extra specification. 

I'm fine if we want to make the match against `key` case-insensitive, but I wouldn't go quite so far as changing the character involved.

Reply:
I could be missing something here but I think code and keyCode are pretty important. I agree that most of the cases you could blame alt as the mapping key but I think that is a valid use case. 

The following are some examples of confusing behavior with just those properties `alt+f`, `alt+n` and `alt+e`. The big issue is with `alt+n` and `alt+e`. In the case of windows you would get key = n/e alt = true. In the case of mac you would get key = dead alt = true for both. The only real way to distinguish these two events on mac is using code (KeyN/KeyE) or keyCode (732/180)

The way I have it now seems like it should handle all cases pretty well. You register the key using its raw value (e/f/n) and any modifiers; However, when you get the event back from the API it provides key in its raw form along with code and keyCode to help you decide how to handle that event.